chia/util/config.py

```diff
@@ -19,9 +19,11 @@ def create_default_chia_config(root_path: Path, filenames=["config.yaml"]) -> No
     for filename in filenames:
         default_config_file_data = initial_config_file(filename)
         path = config_path_for_filename(root_path, filename)
+        tmp_path = path.with_suffix("." + str(os.getpid()))
         mkdir(path.parent)
-        with open(path, "w") as f:
+        with open(tmp_path, "w") as f:
             f.write(default_config_file_data)
+        shutil.move(str(tmp_path), str(path))
```

Comment:
from https://docs.python.org/3/library/shutil.html
> If the destination already exists but is not a directory, it may be overwritten depending on os.rename() semantics.

The documentation for `os.rename()` at https://docs.python.org/3/library/os.html#os.rename says:
> If dst exists, the operation will fail with an OSError subclass in a number of cases:
>
> On Windows, if dst exists a FileExistsError is always raised.

...

> If you want cross-platform overwriting of the destination, use replace().

The documentation for `os.replace()` at https://docs.python.org/3/library/os.html#os.replace says:
> Rename the file or directory src to dst. If dst is a directory, OSError will be raised. If dst exists and is a file, it will be replaced silently if the user has permission. The operation may fail if src and dst are on different filesystems. If successful, the renaming will be an atomic operation (this is a POSIX requirement).

It seems to me that `os.replace()` is a better fit than `shutil.move()`

Reply:
I'll try testing this on Windows. I'll note that this pattern is already used by `save_config()` (just below), which is called much more frequently than `create_default_chia_config()`. If there is an issue with this behavior on Windows, surely it would be manifesting all over the place.

Reply:
@arvidn I've tested this on Windows, and as expected, `shutil.move()` works regardless of whether the destination exists or not. Looking at the implementation of `shutil.move()`, it does call `os.rename()`, but it catches and handles the OSError ([WinError 183] aka FileExistsError) by calling `copyfile()`. Here's the implementation:

https://github.com/python/cpython/blob/84ca5fcd31541929f0031e974a434b95d8e78aab/Lib/shutil.py#L806

Reply:
> it would be good if CI would cover both the case where the file exists and where it doesn't exist. I suspect such test would fail on windows with this patch

I'll add some tests for config.py. `shutil.move()` works fine on Windows, but I agree that `os.replace()` is a better choice.